datadog/resource_datadog_synthetics_test_.go

```diff
@@ -2123,6 +2125,12 @@ func buildDatadogSyntheticsAPITest(d *schema.ResourceData) *datadogV1.Synthetics
 						}
 					}
 				}
+				// Override the request client certificate with the one from the config
+				rawConfig := d.GetRawConfig()
+				configCert, configKey := getConfigCertificate(rawConfig, i)
+				if configCert != nil && configKey != nil {
+					overrideStateCertificate(stepMap["request_client_certificate"].([]interface{}), *configCert, *configKey)
```

Comment:
Let's keep this function pure, just returning the right structure, and overriding here.